openforcefield/topology/molecule.py

```diff
@@ -4590,6 +4591,72 @@ def add_conformer(self, coordinates):
 
         return self._add_conformer(coordinates)
 
+    def visualize(self, backend='rdkit'):
+        """Render a visualization of the molecule in Jupyter"""
+        from openforcefield.utils.toolkits import OPENEYE_AVAILABLE, RDKIT_AVAILABLE
+
+        backend = backend.lower()
+
+        if backend == 'nglview':
+            import nglview as nv
+            if self.conformers:
+                traj = _OFFTrajectoryNGLView(self)
+                widget = nv.NGLWidget()
+                widget.add_trajectory(traj)
+                return widget
+            else:
+                raise ValueError('Visualizing with NGLview requires a molecule has conformers.')
+        elif backend == 'rdkit':
+            if RDKIT_AVAILABLE:
+                from rdkit.Chem.Draw import IPythonConsole
+                return self.to_rdkit()
+        elif backend == 'openeye':
+            if OPENEYE_AVAILABLE:
+                from openeye import oedepict
+                from IPython.display import Image
+
+                oemol = self.to_openeye()
+
+                width, height = 500, 300
+                opts = oedepict.OE2DMolDisplayOptions(width, height, oedepict.OEScale_AutoScale)
+
+                oedepict.OEPrepareDepiction(oemol)
+                img = oedepict.OEImage(width, height)
+                display = oedepict.OE2DMolDisplay(oemol, opts)
+                oedepict.OERenderMolecule(img, display)
+                png = oedepict.OEWriteImageToString("png", img)
+                return Image(png)
+        else:
+            raise ValueError('Could not find an appropriate backend')
+
+# TODO: Gracefull skip this class if nglview is not installed
```

Comment:
Maybe having a conditional class definition works here? I don't know how others feel about it but, maybe:

```
try:
    import nglview as nv
    class _OFFTrajectoryNGLView(nv.Trajectory):
        ... # real class here
except ImportError:
    class _OFFTrajectoryNGLView:
        def __init__(self, *args, **kwargs):
            raise NotImplementedError("You need NGLView to use this class")
```

Reply:
I am tackling this, by the way. Will push later.

Reply:
Check last commits!